src/server/server_family.cc

```diff
@@ -1946,20 +2020,34 @@ void ServerFamily::ReplicaOf(CmdArgList args, ConnectionContext* cntx) {
     return;
   }
 
-  // Flushing all the data after we marked this instance as replica.
-  Transaction* transaction = cntx->transaction;
-  transaction->Schedule();
+  if (flush_db == ShouldFlushDb::kFlush) {
```

Comment:
Maybe rename `flush_db` to `should_flush_db`
(reading this almost 60 lines after the declaration is confusing :)